lib/Versions/VersionsBackend.php

```diff
@@ -67,11 +67,12 @@ private function getFolderIdForFile(FileInfo $file): int {
 
 	public function getVersionFolderForFile(FileInfo $file): Folder {
 		$folderId = $this->getFolderIdForFile($file);
+		$groupfoldersVersionsFolder = $this->getVersionsFolder($folderId);
 
 		try {
-			$groupfoldersVersionsFolder = $this->getVersionsFolder($folderId);
 			/** @var Folder $versionsFolder */
-			return $groupfoldersVersionsFolder->get((string)$file->getId());
+			$versionsFolder = $groupfoldersVersionsFolder->get((string)$file->getId());
+			return $versionsFolder;
```

Comment:
```suggestion
			return $groupfoldersVersionsFolder->get((string)$file->getId());
```

Reply:
This is necessary to assign the type because it returns Node but we know it is Folder and the method expects only Folder to be returned. If not done here it would need to be done in https://github.com/nextcloud/groupfolders/pull/3184 anyway.